core/include/usModule.h

```diff
@@ -1,368 +1,430 @@
+ * An <code>UNLOADED</code> module is a
+ * zombie and can only be reached because it was loaded before. However,
+ * unloaded modules can be loaded again.
+ *
+ * <p>
+ * The framework is the only entity that is allowed to create
+ * <code>%Module</code> objects.
+ *
+ * @remarks This class is thread safe.
+ */
+class US_Core_EXPORT Module
+{
+
+public:
+
+  /**
+   * Returns the property key for looking up this module's id.
+   * The property value is of type \c long.
+   *
+   * @return The id property key.
+   */
+  static const std::string& PROP_ID();
+
+  /**
+   * Returns the property key for looking up this module's name.
+   * The property value is of type \c std::string.
+   *
+   * @return The name property key.
+   */
+  static const std::string& PROP_NAME();
+
+  /**
+   * Returns the property key for looking up this module's
+   * location in the file system.
+   * The property value is of type \c std::string.
+   *
+   * @return The location property key.
+   */
+  static const std::string& PROP_LOCATION();
+
+  /**
+   * Returns the property key with a value of \c module.version for looking
+   * up this module's version identifier.
+   * The property value is of type \c std::string.
+   *
+   * @return The version property key.
+   */
+  static const std::string& PROP_VERSION();
+
+  /**
+   * Returns the property key with a value of \c module.vendor for looking
+   * up this module's vendor information.
+   * The property value is of type \c std::string.
+   *
+   * @return The vendor property key.
+   */
+  static const std::string& PROP_VENDOR();
+
+  /**
+   * Returns the property key with a value of \c module.description for looking
+   * up this module's description.
+   * The property value is of type \c std::string.
+   *
+   * @return The description property key.
+   */
+  static const std::string& PROP_DESCRIPTION();
+
+  /**
+   * Returns the property key with a value of \c module.autoload_dir for looking
+   * up this module's auto-load directory.
+   * The property value is of type \c std::string.
+   *
+   * @return The auto-load directory property key.
+   */
+  static const std::string& PROP_AUTOLOAD_DIR();
+
+  /**
+   * Returns the property key with a value of \c module.autoloaded_modules for
+   * looking up this module's auto-load modules.
+   * The property value is of type \c std::vector<std::string> and contains
+   * the file system locations for the auto-loaded modules triggered by this
+   * module.
+   *
+   * @return The auto-loaded modules property key.
+   */
+  static const std::string& PROP_AUTOLOADED_MODULES();
+
+  virtual ~Module();
+
+  /**
+   * Returns this module's current state.
+   *
+   * <p>
+   * A module can be in only one state at any time.
+   *
+   * @return <code>true</code> if the module is <code>LOADED</code>
+   *         <code>false</code> if it is in any other state.
+   */
+  virtual bool IsLoaded() const;

```

Comment:
I believe this and all the following methods do not need to be virtual. Please investigate.